test/new-e2e/pkg/runner/ci_profile.go

```diff
@@ -73,9 +70,11 @@ func NewCIProfile() (Profile, error) {
 	}, nil
 }
 
-// RootWorkspacePath returns the root directory for CI Pulumi workspace
-func (p ciProfile) RootWorkspacePath() string {
-	return workspaceFolder
+// GetWorkspacePath returns the directory for CI Pulumi workspace.
+// Since one Workspace supports one single program and we have one program per stack,
+// the path should be unique for each stack.
+func (p ciProfile) GetWorkspacePath(stackName string) string {
+	return path.Join(workspaceRootFolder, stackName)
```

Comment:
You may want to sanitize or hash `stackName` to avoid weird paths (esp. on Windows).

Reply:
Good catch !